packages/teleport-project-generator/src/index.ts

```diff
@@ -74,8 +74,14 @@ export const createProjectGenerator = (strategy: ProjectStrategy): ProjectGenera
     // Handling pages, based on the conditionals in the root node
     for (const routeNode of routeNodes) {
       const { files, dependencies } = await createPage(routeNode, strategy, options)
+      const { metaDataOptions } = strategy.pages
 
-      injectFilesToPath(rootFolder, strategy.pages.path, files)
+      const path =
+        metaDataOptions && metaDataOptions.useFolderStructure
```

Comment:
I would move this logic in a function whose name will explain what is happening here